src/browser.js

```diff
@@ -46,68 +46,71 @@ type HydrationStateType = {
 export const HydrationStateToken: Token<HydrationStateType> = createToken(
   'HydrationStateToken'
 );
-const plugin: FusionPlugin<I18nDepsType, I18nServiceType> = createPlugin({
-  deps: {
-    fetch: FetchToken.optional,
-    hydrationState: HydrationStateToken.optional,
-  },
-  provides: ({fetch = window.fetch, hydrationState} = {}) => {
-    class I18n {
-      loadedChunks: Array<number | string>;
-      localeCode: ?string;
-      translationMap: TranslationsObjectType;
 
-      constructor() {
-        const {chunks, localeCode, translations} =
-          hydrationState || loadTranslations();
-        this.loadedChunks = (chunks: Array<number | string> | void) || [];
-        this.localeCode = localeCode;
-        this.translationMap = translations || {};
-      }
-      async load(chunkIds: Array<number | string>): Promise<void> {
-        const unloaded = chunkIds.filter(id => {
-          return this.loadedChunks.indexOf(id) < 0;
-        });
-        const fetchOpts = {
-          method: 'POST',
-          ...(this.localeCode
-            ? {headers: {'X-Fusion-Locale-Code': this.localeCode}}
-            : {}),
-        };
-        if (unloaded.length > 0) {
-          // Don't try to load translations again if a request is already in
-          // flight. This means that we need to add unloaded chunks to
-          // loadedChunks optimistically and remove them if some error happens
-          this.loadedChunks = [...this.loadedChunks, ...unloaded];
+type PluginType = FusionPlugin<I18nDepsType, I18nServiceType>;
+const pluginFactory: () => PluginType = () =>
```

Comment:
Happy to hear thoughts on alternatives here until tree shaking can catch the case(s) on master.

Benefit of a function here is that it does not evaluate until after the `__BROWSER__` check below.  I personally find this more clear as to why it tree shakes correctly.